release/pipeline/dags/istio_common_dag.py

```diff
@@ -65,136 +76,91 @@ def GetSettingTemplate(setting):
-    # What version number to use for this round.
-    r_minor = int(AirflowGetVariableOrBaseCase('released_version_minor', 0))
-    r_patch = int(AirflowGetVariableOrBaseCase('released_version_patch', 0))
-    # If  we have already released a monthy for this mounth then bump
-    # The patch number for the remander of the month.
-    if r_minor == minor_version:
-      patch = r_patch + 1
-    else:
-      patch = 0
-    # If version is overriden then we should use it otherwise we use it's
-    # default or monthly value.
-    version = conf.get('VERSION')
-    if monthly and not version:
-      version = '{}.{}.{}'.format(major_version, minor_version, patch)
-
-    default_conf = environment_config.get_airflow_config(
-        version,
-        timestamp,
-        major=major_version,
-        minor=minor_version,
-        patch=patch,
-        date=date.strftime('%Y%m%d'),
-        rc=date.strftime('%H-%M'))
-    config_settings = dict(VERSION=default_conf['VERSION'])
-    config_settings_name = [
-        'PROJECT_ID',
-        'MFEST_URL',
-        'MFEST_FILE',
-        'GCS_STAGING_BUCKET',
-        'SVC_ACCT',
-        'GITHUB_ORG',
-        'GITHUB_REPO',
-        'GCS_GITHUB_PATH',
-        'TOKEN_FILE',
-        'GCR_STAGING_DEST',
-        'GCR_RELEASE_DEST',
-        'GCS_MONTHLY_RELEASE_PATH',
-        'DOCKER_HUB',
-        'GCS_BUILD_BUCKET',
-        'RELEASE_PROJECT_ID',
-    ]
-
-    for name in config_settings_name:
-      config_settings[name] = conf.get(name) or default_conf[name]
-
-    if monthly:
-      config_settings['MFEST_COMMIT'] = conf.get(
-          'MFEST_COMMIT') or Variable.get('latest_sha')
-      gcs_path = conf.get('GCS_MONTHLY_STAGE_PATH')
-      if not gcs_path:
-        gcs_path = default_conf['GCS_MONTHLY_STAGE_PATH']
-    else:
-      config_settings['MFEST_COMMIT'] = conf.get(
-          'MFEST_COMMIT') or default_conf['MFEST_COMMIT']
-      gcs_path = conf.get('GCS_DAILY_PATH') or default_conf['GCS_DAILY_PATH']
-
-    config_settings['GCS_STAGING_PATH'] = gcs_path
-    config_settings['GCS_BUILD_PATH'] = '{}/{}'.format(
-        config_settings['GCS_BUILD_BUCKET'], gcs_path)
-    config_settings['GCS_RELEASE_TOOLS_PATH'] = '{}/release-tools/{}'.format(
-        config_settings['GCS_BUILD_BUCKET'], gcs_path)
-    config_settings['GCS_FULL_STAGING_PATH'] = '{}/{}'.format(
-        config_settings['GCS_STAGING_BUCKET'], gcs_path)
-    config_settings['ISTIO_REPO'] = 'https://github.com/{}/{}.git'.format(
-        config_settings['GITHUB_ORG'], config_settings['GITHUB_REPO'])
-
-    return config_settings
+  tasksOD = collections.OrderedDict(CommonTasks)
 
   generate_flow_args = PythonOperator(
       task_id='generate_workflow_args',
-      python_callable=GenerateTestArgs,
+      python_callable=dag_args_func,
       provide_context=True,
       dag=common_dag,
   )
+  tasksOD['generate_workflow_args'] = generate_flow_args
 
   get_git_commit_cmd = """
```

Comment:
How is this even test-able? Can we move these commands to separate .sh file, and read them in python?